File: app/src/main/java/com/ak47/www/koko_androiddemo/sqllite/WaitlistDbHelper.java
package com.ak47.www.koko_androiddemo.sqllite;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;
import android.util.Log;

public class WaitlistDbHelper extends SQLiteOpenHelper {
    private static final String DATABASE_NAME = "waitlist.db";
    private static final int DATEBASE_VERSION = 1;

    public WaitlistDbHelper(Context context) {
        super(context, DATABASE_NAME, null, DATEBASE_VERSION);
    }


    @Override
    public void onCreate(SQLiteDatabase sqLiteDatabase) {
        //很容易忘记必要的空格 "CREATE TABLE "
        //很容易忘记分割的 ，
        //很容易忘记结尾 ；
        final String SQL_CREATE_WAITLIST_TABLE = "CREATE TABLE " + WaitlistContract.WaitlistEntry.TABLE_NAME + " (" +
                WaitlistContract.WaitlistEntry._ID + " INTEGER PRIMARY KEY AUTOINCREMENT," +
                WaitlistContract.WaitlistEntry.COLUMN_PARTY_SIZE + " INTEGER NOT NULL, " +
                WaitlistContract.WaitlistEntry.COLUMN_GUEST_NAME + " TEXT NOT NULL, " +
                WaitlistContract.WaitlistEntry.COLUMN_TIMESTAMP + " TIMESTAMP DEFAULT CURRENT_TIMESTAMP" +
                "); ";
        Log.e("TAG", SQL_CREATE_WAITLIST_TABLE.toString());

        sqLiteDatabase.execSQL(SQL_CREATE_WAITLIST_TABLE);
    }

    @Override
    public void onUpgrade(SQLiteDatabase sqLiteDatabase, int i, int i1) {
        sqLiteDatabase.execSQL("DROP TABLE IF EXISTS " + WaitlistContract.WaitlistEntry.TABLE_NAME);
        onCreate(sqLiteDatabase);
    }
}

File: app/src/main/java/com/koko/www/androiddemo/sqlite/WaitListDbHelper.kt
package com.koko.www.androiddemo.sqlite

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper

import com.koko.www.androiddemo.sqlite.WaitListContract.WaitListEntry.Companion.COLUMN_GUEST_NAME
import com.koko.www.androiddemo.sqlite.WaitListContract.WaitListEntry.Companion.COLUMN_PARTY_SIZE
import com.koko.www.androiddemo.sqlite.WaitListContract.WaitListEntry.Companion.COLUMN_TIMESTAMP
import com.koko.www.androiddemo.sqlite.WaitListContract.WaitListEntry.Companion.ID
import com.koko.www.androiddemo.sqlite.WaitListContract.WaitListEntry.Companion.TABLE_NAME

class WaitListDbHelper(context: Context) : SQLiteOpenHelper(context, DATABASE_NAME, null, DATABASE_VERSION) {


    override fun onCreate(sqLiteDatabase: SQLiteDatabase) {
        //很容易忘记必要的空格 "CREATE TABLE "
        //很容易忘记分割的 ，
        //很容易忘记结尾 ；
        val sqlCreateWaitListTable = "CREATE TABLE " + TABLE_NAME + " (" +
                ID + " INTEGER PRIMARY KEY AUTOINCREMENT," +
                COLUMN_PARTY_SIZE + " INTEGER NOT NULL, " +
                COLUMN_GUEST_NAME + " TEXT NOT NULL, " +
                COLUMN_TIMESTAMP + " TIMESTAMP DEFAULT CURRENT_TIMESTAMP" +
                "); "
        sqLiteDatabase.execSQL(sqlCreateWaitListTable)
    }

    override fun onUpgrade(sqLiteDatabase: SQLiteDatabase, i: Int, i1: Int) {
        sqLiteDatabase.execSQL("DROP TABLE IF EXISTS $TABLE_NAME")
        onCreate(sqLiteDatabase)
    }

    companion object {
        private const val DATABASE_NAME = "waitList.db"
        private const val DATABASE_VERSION = 2
    }
}
